### PR TITLE
shells now also given in matrices

### PR DESCRIPTION
**Pull request by samwaseda**

### CODE REVIEW

**Comment by jan-janssen** [comment]
@samwaseda It would be nice to add some tests for these new functions.

**Comment by sudarsan-surendralal** [comment]
Regarding tests, check out the tests at https://github.com/pyiron/pyiron_atomistics/blob/master/tests/structure/test_atoms.py. It's just functions where you call the function you wrote to test if the output of that function works the way you want it to (compare the values and the type of output)

In your case, it would be useful to define something like

```
def test_get_shell_matrix(self
      output = basis.get_shell_matrix()
      self.assertIstype(output, np.ndarray)
```

**Comment by samwaseda** [comment]
Done!

**Comment by jan-janssen** [comment]
The code does not seem to be Python2 compatible: 

```
======================================================================
ERROR: test_get_shell_matrix (structure.test_atoms.TestAtoms)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pyiron/pyiron_atomistics/tests/structure/test_atoms.py", line 377, in test_get_shell_matrix
    output = basis.get_shell_matrix(shell=1)
  File "pyiron_atomistics/structure/atoms.py", line 1493, in get_shell_matrix
    shell_matrix[ii][neigh_list.indices[ii, ss==shell]] = 1
TypeError: list indices must be integers, not tuple
----------------------------------------------------------------------
Ran 106 tests in 24.867s
FAILED (errors=1)
```

File: pyiron_atomistics/structure/atoms.py [hunk 1]
```diff
@@ -1490,7 +1490,7 @@ def get_shell_matrix(self, shell, neigh_list=None, id_list=None, restraint_matri
             restraint_matrix = ((restraint_matrix+restraint_matrix.transpose())>0)
         shell_matrix = np.zeros((Natom,Natom))
         for ii, ss in enumerate(neigh_list.shells):
-            shell_matrix[ii][neigh_list.indices[ii, ss==shell]] = 1
+            shell_matrix[ii][neigh_list.indices[ii, ss==np.array(shell)]] = 1
```
Comment (jan-janssen) [inline pyiron_atomistics/structure/atoms.py]: The error remains: 
```
======================================================================
ERROR: test_get_shell_matrix (structure.test_atoms.TestAtoms)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pyiron/pyiron_atomistics/tests/structure/test_atoms.py", line 377, in test_get_shell_matrix
    output = basis.get_shell_matrix(shell=1)
  File "pyiron_atomistics/structure/atoms.py", line 1493, in get_shell_matrix
    shell_matrix[ii][neigh_list.indices[ii, ss==np.array(shell)]] = 1
TypeError: list indices must be integers, not tuple
----------------------------------------------------------------------
```

Comment (samwaseda) [reply pyiron_atomistics/structure/atoms.py]: Actually I don't really understand what's the problem...

**Pull request merged**